pkg/node/manager/manager.go

```diff
@@ -334,8 +334,12 @@ func (m *Manager) NodeUpdated(n nodeTypes.Node) {
 	nodeIP := n.GetNodeIP(false)
 
 	remoteHostIdentity := identity.ReservedIdentityHost
-	if m.conf.RemoteNodeIdentitiesEnabled() && n.Source != source.Local {
-		remoteHostIdentity = identity.ReservedIdentityRemoteNode
+	if m.conf.RemoteNodeIdentitiesEnabled() {
+		if n.NodeIdentity != 0 {
+			remoteHostIdentity = identity.NumericIdentity(n.NodeIdentity)
+		} else if n.Source != source.Local {
+			remoteHostIdentity = identity.ReservedIdentityRemoteNode
+		}
```

Comment:
```suggestion
		ni := identity.NumericIdentity(n.NodeIdentity)
		if ni != identity.IdentityUnknown {
			remoteHostIdentity = ni
		} else if n.Source != source.Local {
			remoteHostIdentity = identity.ReservedIdentityRemoteNode
		}
```

Reply:
Ok, see the value of casting before comparing, changed :-)